File: quan.java/database/src/test/java/quan/data/role/RoleType.java
package quan.data.role;

/**
 * 角色类型<br/>
 * 自动生成
 */
public enum RoleType {

    /**
     * 角色类型1
     */
    type1(1),

    /**
     * 角色类型2
     */
    type2(2);


    private int value;

    RoleType(int value) {
        this.value = value;
    }

    public int value() {
        return value;
    }

    public static RoleType valueOf(int value) {
        switch (value) {
            case 1:
                return type1;
            case 2:
                return type2;
            default:
                return null;
        }
    }

}

File: quan.lua/src/test/message/role/RoleType.lua
require("quan.table")

---
---角色类型
---自动生成，请勿修改
---
local RoleType = {
    --- 角色类型1
    type1 = 1,
    --- 角色类型2
    type2 = 2
}

RoleType = table.readOnly(RoleType)
return RoleType